### PR TITLE
Cleanup callbacks implementation in CodyAgentClient

### PR DESCRIPTION
**Pull request by pkukielka**
## Changes

Our callbacks logic in the `CodyAgentClient` is unnecessary complicated. By making `project` a class parameter we can avoid a lot of superfluous null handling.

## Test plan

N/A

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentClient.kt [hunk 1]
```diff
@@ -22,156 +30,114 @@ import org.eclipse.lsp4j.jsonrpc.services.JsonRequest
  * Implementation of the client part of the Cody agent protocol. This class dispatches the requests
  * and notifications sent by the agent.
  */
-@Suppress("unused")
-class CodyAgentClient(private val webview: NativeWebviewProvider) {
+@Suppress("unused", "FunctionName")
+class CodyAgentClient(private val project: Project, private val webview: NativeWebviewProvider) {
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentClient.kt:33]: hah, did the same in https://github.com/sourcegraph/jetbrains/pull/2040 😃

File: src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentClient.kt [hunk 2]
```diff
@@ -229,28 +195,17 @@ class CodyAgentClient(private val webview: NativeWebviewProvider) {
   // TODO: Remove this
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentClient.kt]: when do we want to remove this?

File: src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentClient.kt [hunk 3]
```diff
@@ -229,28 +195,17 @@ class CodyAgentClient(private val webview: NativeWebviewProvider) {
...
-      }
-      return
+    if (params.message.type == ExtensionMessage.Type.SET_CONFIG_FEATURES) {
+      runInEdt { onSetConfigFeatures.update(params.message.configFeatures) }
     }
 
     logger.debug("webview/postMessage ${params.id}: ${params.message}")
   }
+
+  // TODO: Remove this
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentClient.kt]: why keeping it?

**Review by mkondratek — APPROVED** [review]
lgtm

File: src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentClient.kt (hunk 2 above)
Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentClient.kt]: Removed

File: src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentClient.kt (hunk 3 above)
Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentClient.kt]: Removed

**Pull request merged**